src/Servers/IIS/AspNetCoreModuleV2/IISLib/stringa.cpp

```diff
@@ -1166,28 +1169,23 @@ STRA::AuxAppendWTruncate(
 // Cheesey WCHAR --> CHAR conversion
 //
 {
-    HRESULT hr = S_OK;
-
     _ASSERTE( NULL != pszAppendW );
     _ASSERTE( 0 == cbOffset || cbOffset == QueryCB() );
 
     if( !pszAppendW )
     {
-        hr = HRESULT_FROM_WIN32( ERROR_INVALID_PARAMETER );
-        goto Finished;
+        return HRESULT_FROM_WIN32( ERROR_INVALID_PARAMETER );
     }
 
     ULONGLONG cbNeeded = static_cast<ULONGLONG>(cbOffset) + cchAppendW + sizeof( CHAR );
     if( cbNeeded > MAXDWORD )
     {
-        hr = HRESULT_FROM_WIN32( ERROR_ARITHMETIC_OVERFLOW );
-        goto Finished;
+        return HRESULT_FROM_WIN32( ERROR_ARITHMETIC_OVERFLOW );
     }
 
     if( !m_Buff.Resize( static_cast<SIZE_T>(cbNeeded) ) )
     {
-        hr = E_OUTOFMEMORY;
-        goto Finished;
```

Comment:
Wow gotos!

Reply:
Sometimes it's C code spiffed up in a .cpp file extension. :)